pvlib/tracking.py

```diff
@@ -569,3 +582,163 @@ def singleaxis(apparent_zenith, apparent_azimuth,
+    system_zenith : float
+        tilt of normal to slope relative to vertical [radians]
+    tracker_azimuth : float
+        direction of tracker axes on horizontal [radians]
+
+    Returns
+    -------
+    tracker_zenith : float
+        tilt of tracker [radians]
+
+    Solving for the tracker tilt on a slope is derived in the following steps:
+
+    1. the trackers axes are in the system plane, so the ``z-coord = 0``
+
+    2. rotate the trackers ``[x_tr_sys, y_tr_sys, 0]`` back to the global, but
+       rotated by the tracker global azimuth if there is one, so that the
+       tracker axis is constrained to y-z plane so that ``x-coord = 0`` ::
+
+        Rx_sys = [[1,           0,            0],
+                  [0, cos(sys_ze), -sin(sys_ze)],
+                  [0, sin(sys_ze),  cos(sys_ze)]]
+
+        Rz_sys = [[cos(sys_az-tr_az), -sin(sys_az-tr_az), 0],
+                  [sin(sys_az-tr_az),  cos(sys_az-tr_az), 0],
+                  [                0,                  0, 1]]
+
+        tr_rot_glo = Rz_sys.T * (Rx_sys.T * [x_tr_sys, y_tr_sys, 0])
+
+        tr_rot_glo = [
+          [ x_tr_sys*cos(sys_az-tr_az)+y_tr_sys*sin(sys_az-tr_az)*cos(sys_ze)],
+          [-x_tr_sys*sin(sys_az-tr_az)+y_tr_sys*cos(sys_az-tr_az)*cos(sys_ze)],
+          [                                             -y_tr_sys*sin(sys_ze)]]
+
+    3. solve for ``x_tr_sys`` ::
+
+        x_tr_sys*cos(sys_az-tr_az)+y_tr_sys*sin(sys_az-tr_az)*cos(sys_ze) = 0
+        x_tr_sys = -y_tr_sys*tan(sys_az-tr_az)*cos(sys_ze)
+
+    4. so tracker axis tilt, ``tr_ze = arctan2(tr_rot_glo_z, tr_rot_glo_y)`` ::
+
+        tr_rot_glo_y = y_tr_sys*cos(sys_ze)*(
+          tan(sys_az-tr_az)*sin(sys_az-tr_az) + cos(sys_az-tr_az))
+
+        tan(tr_ze) = -y_tr_sys*sin(sys_ze) / tr_rot_glo_y
+
+    The trick is multiply top and bottom by cos(sys_az-tr_az) and remember that
+    ``sin^2 + cos^2 = 1`` (or just use sympy.simplify) ::
+
+        tan(tr_ze) = -tan(sys_ze)*cos(sys_az-tr_az)
+    """
+    sys_az_rel_to_tr_az = system_azimuth - tracker_azimuth
+    tan_tr_ze = -np.cos(sys_az_rel_to_tr_az) * np.tan(system_zenith)
+    return -np.arctan(tan_tr_ze)
+
+
+def calc_system_tracker_side_slope(
+        tracker_azimuth, tracker_zenith, system_azimuth, system_zenith):
+    """
+    Calculate the slope perpendicular to the tracker axis relative to the
+    system plane containing the axes as well as the rotation of the tracker
+    axes relative to the system plane. Note in order for the backtracking
+    algorithm to work correctly on a sloped system plane, the side slope must
+    be applied to the tracker rotation.
+
+    Parameters
+    ----------
+    system_azimuth : float
+        direction of normal to slope on horizontal [radians]
+    system_zenith : float
+        tilt of normal to slope relative to vertical [radians]
+    tracker_azimuth : float
+        direction of tracker axes on horizontal [radians]
+    tracker_zenith : float
+        tilt of tracker [radians]
+
```

Comment:
same comments as above about parameter naming and units

Reply:
change `tracker_azimuth` to `axis_azimuth` and `tracker_zenith` to `axis_tilt`, and changed all units to degrees. Still thinking about whether "slope" is better than "system"